src/entities/friendship_history.rs

```diff
@@ -131,9 +134,12 @@ impl FriendshipHistoryRepository {
         }
     }
 
-    fn get_executor<'a>(&'a self, transaction: Option<Transaction<'a, Postgres>>) -> Executor<'a> {
+    fn get_executor(
+        &self,
+        transaction: Option<Transaction<'static, Postgres>>,
+    ) -> Executor<'static> {
         transaction.map_or_else(
-            || Executor::Pool(DatabaseComponent::get_connection(&self.db_connection)),
+            || Executor::Pool(DatabaseComponent::get_connection(&self.db_connection).clone()), // Clone because it's cheap and the pool use an Arc internally
```

Comment:
nit: "// Choose to `clone` because it's cheap and the pool uses an Arc internally"